### PR TITLE
chore: Remove manual mocks from tests

### PR DESCRIPTION
**Pull request by marikaner**
We still had a left-over from ancient times, where we mocked fs manually. Using `mock-fs` instead.

### CODE REVIEW

File: packages/test-util/src/test-destination-provider.ts [hunk 1]
```diff
@@ -244,7 +244,7 @@ function readCredentials(filePath: string): CredentialsFile {
 function readJson(filePath: string): { [key: string]: any } {
   let content;
   try {
-    content = readFileSync(filePath, 'utf8');
+    content = readFileSync(filePath, 'utf-8');
```
Comment (tomfrenken) [inline packages/test-util/src/test-destination-provider.ts:247]: Might as well change it to async 👀

File: packages/test-util/src/test-destination-provider.ts [hunk 2]
```diff
@@ -167,7 +167,7 @@ function readCredentialsOrFail(
 function mergeSystemsAndCredentials(systems, credentials) {
   return systems.map(system => ({
     ...system,
-    ...credentials.find(cred => cred.alias === system.alias)
+    ...credentials.find(({ alias }) => alias === system.alias)
```
Comment (tomfrenken) [inline packages/test-util/src/test-destination-provider.ts:170]: Didn't know you can destructure from a spread object like this, that's cool!

**Review by tomfrenken — APPROVED** [review]
LGTM, one minor suggestions though.

File: packages/test-util/src/test-destination-provider.ts (hunk 1 above)
Comment (marikaner) [reply packages/test-util/src/test-destination-provider.ts:247]: No, because this would change the public API. ¯\_(ツ)_/¯

Comment (tomfrenken) [reply packages/test-util/src/test-destination-provider.ts:247]: Indeed 😬 - I just saw `test` in the file path and my brain jumped to private API.

**Pull request merged**